hp_list/main.c

```diff
@@ -14,6 +14,140 @@
 #include <string.h>
 #include <threads.h>
 
+#ifdef RUNTIME_STAT
+/*
+ * "rtry" the number of retries in the __list_find function.
+ * "cons" the number of wait-free contains in the __list_find function that curr
+ * pointer pointed.
+ * "trav" the number of list element traversal in the __list_find function.
+ * "fail" the number of CAS() failures.
+ * "del" the number of list_delete operation failed and restart again.
+ * "ins" the number of list_insert operation failed and restart again.
+ * "load"  is the number of atomic_load operation in list_delete, list_insert
+ * and __list_find.
+ * "store" is the number of atomic_store operation in list_delete, list_insert
+ * and __list_find.
+ */
+static atomic_uint_fast64_t rtry = 0, cons = 0, trav = 0, fail = 0;
+static atomic_uint_fast64_t del = 0, ins = 0;
+static atomic_uint_fast64_t load = 0, store = 0;
+static atomic_uint_fast64_t deletes = 0, inserts = 0;
+
+#define TRACE_GOTO_TRY_AGAIN        \
+    do {                            \
+        atomic_fetch_add(&rtry, 1); \
+        goto try_again;             \
```

Comment:
Don't do this!
Let's simply the effectiveness of macro `TRACE_GOTO_TRY_AGAIN`.
It should appear before the statement `goto try_again;` before it is about is be expanded.